utils/build.py

```diff
@@ -0,0 +1,173 @@
+# limitations under the License.
+
+"""
+Builds the Shaderc project, on Linux, Mac, or Windows.
+"""
+
+from __future__ import print_function
+import argparse
+import os
+import platform
+import subprocess
+
+
+OS = platform.system()
+
+
+def run(cmd, cwd, justprint):
+    """Prints a command to run, and optionally runs it.
+
+    Raises a RuntimeError if the command does not launch or otherwise fails.
+
+    Args:
+      justprint: If true, then only print the command. Otherwise run the 
+                   command after printing it.
+      cmd:       List of words in the command.
+      cwd:       Working directory for the command.
+    """
+    print(cmd)
+    if justprint:
+        return
+
+    p = subprocess.Popen(cmd, cwd=cwd)
+    (_, _) = p.communicate()
+    if p.returncode != 0:
+        raise RuntimeError("Failed to run %s in %s" % (cmd, cwd))
+
+
+def quote(string):
+    return '"%s"' % string.replace('"', '\\"')
+
+
+def cmake_batline(cmake_command):
+    """Transforms cmake_command (a list of strings) into a single string
+    suitable for inserting into a .bat script.
+    """
+    args_to_quote = [0, 1, 4]
+    quoted =[ (quote(e) if i in args_to_quote else e)
+              for (i, e) in enumerate(cmake_command) ]
+    return ' '.join(quoted)
+
+
+def build(args):
+    """ Builds Shaderc under specified conditions.
+
+    Args:
+        args: An object with attributes:
+            cmake: path to the cmake executable
+            ninja: path to the ninja executable
+            srcdir: where Shaderc source can be found
+            builddir: build working directory
+            installdir: install directory
+    """
+
+    if not os.path.isdir(args.srcdir):
+        raise RuntimeError("Soure directory %s does not exist" % (args.srcdir))
+
+    # Make paths absolute, and ensure directories exist.
+    for d in [args.builddir, args.installdir]:
+        if not os.path.isdir(d):
+            os.makedirs(d)
+    cmake = (os.path.abspath(args.cmake)
+             if os.path.isfile(args.cmake) else 'cmake')
+    ctest = 'ctest' if cmake == 'cmake' else os.path.join(
+        os.path.dirname(cmake), 'ctest')
+    ninja = (os.path.abspath(args.ninja)
+             if os.path.isfile(args.ninja) else 'ninja')
+    args.srcdir = os.path.abspath(args.srcdir)
+    args.builddir = os.path.abspath(args.builddir)
+    args.installdir = os.path.abspath(args.installdir)
+
+    print('Building Shaderc:')
+    print('   Source     : ', args.srcdir)
+    print('   Cmake      : ', cmake)
+    print('   Ninja      : ', ninja)
+    print('   Build dir  : ', args.builddir)
+    print('   Install dir: ', args.installdir)
+    cmake_command = [cmake, args.srcdir, '-GNinja',
+                     '-DCMAKE_BUILD_TYPE=%s' % args.buildtype,
+                     '-DCMAKE_INSTALL_PREFIX=%s' % (args.installdir)]
+
+    if (OS == 'Windows' or OS.startswith('CYGWIN')):
+        bat_content = '''
+call "%VS140COMNTOOLS%..\..\VC\\vcvarsall.bat"
```

Comment:
Please verify that running vcvarsall without arguments gives you a compiler that will target 32-bit x86.